src/App.test.jsx

```diff
@@ -1,8 +1,242 @@
+    launchYear: 2021,
+    types: ['Public'],
+  },
+  {
+    date: '2022-07-04',
+    localName: 'Independence Day',
+    name: 'Independence Day',
+    countryCode: 'US',
+    fixed: false,
+    global: true,
+    counties: null,
+    launchYear: null,
+    types: ['Public'],
+  },
+  {
+    date: '2022-09-05',
+    localName: 'Labor Day',
+    name: 'Labour Day',
+    countryCode: 'US',
+    fixed: false,
+    global: true,
+    counties: null,
+    launchYear: null,
+    types: ['Public'],
+  },
+  {
+    date: '2022-10-10',
+    localName: 'Columbus Day',
+    name: 'Columbus Day',
+    countryCode: 'US',
+    fixed: false,
+    global: false,
+    counties: [
+      'US-AL',
+      'US-AZ',
+      'US-CO',
+      'US-CT',
+      'US-DC',
+      'US-GA',
+      'US-ID',
+      'US-IL',
+      'US-IN',
+      'US-IA',
+      'US-KS',
+      'US-KY',
+      'US-LA',
+      'US-ME',
+      'US-MD',
+      'US-MA',
+      'US-MS',
+      'US-MO',
+      'US-MT',
+      'US-NE',
+      'US-NH',
+      'US-NJ',
+      'US-NM',
+      'US-NY',
+      'US-NC',
+      'US-OH',
+      'US-OK',
+      'US-PA',
+      'US-RI',
+      'US-SC',
+      'US-TN',
+      'US-UT',
+      'US-VA',
+      'US-WV',
+    ],
+    launchYear: null,
+    types: ['Public'],
+  },
+  {
+    date: '2022-11-11',
+    localName: 'Veterans Day',
+    name: 'Veterans Day',
+    countryCode: 'US',
+    fixed: false,
+    global: true,
+    counties: null,
+    launchYear: null,
+    types: ['Public'],
+  },
+  {
+    date: '2022-11-24',
+    localName: 'Thanksgiving Day',
+    name: 'Thanksgiving Day',
+    countryCode: 'US',
+    fixed: false,
+    global: true,
+    counties: null,
+    launchYear: 1863,
+    types: ['Public'],
+  },
+  {
+    date: '2022-12-26',
+    localName: 'Christmas Day',
+    name: 'Christmas Day',
+    countryCode: 'US',
+    fixed: false,
+    global: true,
+    counties: null,
+    launchYear: null,
+    types: ['Public'],
+  },
+];
+
+const server = setupServer(
+  rest.get(
+    'https://date.nager.at/api/v3/PublicHolidays/2022/US',
+    (req, res, ctx) => {
+      return res(ctx.json(mockResponse));
+    }
+  )
+);
+
+beforeAll(() => server.listen());
+
+afterAll(() => server.close());
+
+jest.setTimeout(7000);
+test('should display user search returning one matching item correctly', async () => {
+  render(<App />);
+  const search = await screen.findByRole('textbox');
+  const button = await screen.findByRole('button');
+  userEvent.type(search, `New Year's Day`);
+  userEvent.click(button);
+  await waitFor(() => expect(screen.getAllByRole('heading')).toHaveLength(4), {
+    timeout: 5000,
+  });
+});
+
+test('should display multiples if matches user search', async () => {
+  render(<App />);
+  const search = await screen.findByRole('textbox');
+  const button = await screen.findByRole('button');
+  userEvent.type(search, `in`);
+  userEvent.click(button);
+  await waitFor(() => expect(screen.getAllByRole('heading')).toHaveLength(13), {
```

Comment:
This `waitFor `for may not be necessary now that you are using MSW to mock your API's response